File: yandex_backend/40. Много стульев/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public static void Main(string[] args)
    {
        string[] input = Console.ReadLine().Split();
        int n = int.Parse(input[0]);
        int m = int.Parse(input[1]);

        int[] sellers = Console.ReadLine().Split().Select(int.Parse).ToArray();
        int[] buyers = Console.ReadLine().Split().Select(int.Parse).ToArray();

        Array.Sort(sellers);
        Array.Sort(buyers);

        long profit = 0;
        int sellerIdx = 0;
        int buyerIdx = m - 1;

        while (sellerIdx < n && buyerIdx >= 0)
        {
            if (buyers[buyerIdx] > sellers[sellerIdx])
            {
                profit += (long)buyers[buyerIdx] - sellers[sellerIdx];
                sellerIdx++;
                buyerIdx--;
            }
            else
            {
                buyerIdx--;
            }
        }

        Console.WriteLine(profit);
    }
}

File: yandex_backend/40. Много стульев/solution_.py
import sys
from typing import List

def main() -> None:
    n, m = map(int, input().split())
    sellers: List[int] = list(map(int, input().split()))
    buyers: List[int] = list(map(int, input().split()))

    sellers.sort()
    buyers.sort()

    profit: int = 0
    seller_idx: int = 0
    buyer_idx: int = m - 1

    while seller_idx < n and buyer_idx >= 0:
        if buyers[buyer_idx] > sellers[seller_idx]:
            profit += buyers[buyer_idx] - sellers[seller_idx]
            seller_idx += 1
            buyer_idx -= 1
        else:
            buyer_idx -= 1

    print(profit)


if __name__ == '__main__':
    main()